$OpenBSD: patch-files_ml,v 1.1.1.1 2005/05/22 17:44:41 sturm Exp $
Fix the problem with comparison of files using diff
when the files are on two different hosts.
--- files.ml.orig	Sat Mar 12 11:17:59 2005
+++ files.ml	Sun May 22 08:13:59 2005
@@ -476,7 +476,7 @@ let rec diff root1 path1 ui1 root2 path2
              (Update.translatePath root2 path2 >>= (fun path2 ->
               Copy.file root2 path2 root1 workingDir tmppath realPath
                 `Copy (Props.setLength Props.fileSafe (Props.length desc2))
-                 fp1 ress1 id));
+                 fp2 ress2 id));
            displayDiff
 	     (Fspath.concat workingDir realPath)
              (Fspath.concat workingDir tmppath);
@@ -494,7 +494,7 @@ let rec diff root1 path1 ui1 root2 path2
               (* Note that we don't need the ressource fork *)
               Copy.file root1 path1 root2 workingDir tmppath realPath
                 `Copy (Props.setLength Props.fileSafe (Props.length desc1))
-                 fp2 ress2 id));
+                 fp1 ress1 id));
            displayDiff
              (Fspath.concat workingDir tmppath)
 	     (Fspath.concat workingDir realPath);
